src/ProductConstructionService/ProductConstructionService.Api/Queue/PcsJobsProcessor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.Text.Json;
+using System.Threading;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using ProductConstructionService.Api.Queue.WorkItems;
+
+namespace ProductConstructionService.Api.Queue;
+
+public class PcsJobsProcessor(
+    ILogger<PcsJobsProcessor> logger,
+    PcsJobProcessorOptions options,
+    PcsJobsProcessorStatus status,
+    QueueServiceClient queueServiceClient)
+    : BackgroundService
+{
+    private readonly ILogger<PcsJobsProcessor> _logger = logger;
+    private readonly PcsJobProcessorOptions _options = options;
+    private readonly PcsJobsProcessorStatus _status = status;
+
+    private const int JobInvisibilityTimeoutSeconds = 30;
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        await Task.Run(async () => {
```

Comment:
Why are we inside `Task.Run`?

Reply:
We want to run this in a separate thread. I fixed it so it aligns with what the documentation says